src/step.rs

```diff
@@ -1,22 +1,24 @@
 use core::cmp::Ordering;
 
+/// Step of the consensus protocol.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Ord)]
 #[repr(u8)]
-pub enum State {
-    Reject = 0x00,
-    NewRound = 0x01,
-    Propose = 0x02,
-    Prevote = 0x03,
-    Precommit = 0x04,
-    Commit = 0x05,
+pub enum Step {
+    /// A round just started without a proposal from a leader.
+    NewRound = 0x00,
+    /// The block proposal from the leader was accepted in the network.
+    Propose = 0x01,
+    /// The block acceptance is ready to commit in the network.
+    Prevote = 0x02,
+    /// The commit is performed locally and should be accepted by the peers.
+    Precommit = 0x03,
+    /// The round is finalized with a commit.
+    Commit = 0x04,
 }
 
-impl PartialOrd for State {
+impl PartialOrd for Step {
```

Comment:
Not sure why we need custom PartialOrd as we can just derive it?

Reply:
This defines the consensus order. Reject is a special case, and we can easily follow what happens internally with the steps if we just check this implementation. If we just derive, then the rule isn't clearly written in the code - it is, instead, magic

Reply:
`Reject` was removed and `PartialOrd` would work the same as derived one, or I am missing something. And with `C` like enums wouldn't you already know what to expect from `PartialOrd`?